src/components/measureEditor/MeasureEditor.tsx

```diff
@@ -78,18 +80,152 @@ const MeasureEditor = () => {
   const { getUserName } = useOktaTokens();
   const userName = getUserName();
   const canEdit = userName === measure.createdBy;
+  const terminologyServiceApi = useTerminologyServiceApi();
+  const [valuesetMsg, setValuesetMsg] = useState(null);
+
+  const getValueSetErrors = async (
+    valuesetsArray: ElmValueSet[],
+    tgtValue: string
+  ): Promise<ElmTranslationError[]> => {
+    const valuesetsErrorArray: ElmTranslationError[] = [];
+    if (valuesetsArray && tgtValue) {
+      const results = await Promise.allSettled(
+        valuesetsArray.map(async (valueSet, i) => {
+          const oid = getOid(valueSet);
+          await terminologyServiceApi
+            .getValueSet(tgtValue, oid, valueSet.locator)
+            .then((response) => {
+              if (response.errorMsg) {
+                const vsErrorForElmTranslationError: ElmTranslationError =
+                  processValueSetErrorForElmTranslationError(
+                    response.errorMsg.toString(),
+                    valueSet.locator
+                  );
+                valuesetsErrorArray.push(vsErrorForElmTranslationError);
+              }
+            });
+        })
+      );
+      return valuesetsErrorArray;
+    }
+  };
+
+  const getTgt = (): any => {
+    return window.localStorage.getItem("TGT");
+  };
+  const getTgtValue = (tgt: any): any => {
+    let tgtValue = null;
+    if (tgt) {
+      let tgtObjFromLocalStorage = JSON.parse(tgt);
+      for (const [key, value] of Object.entries(tgtObjFromLocalStorage)) {
+        if (key === "TGT") {
+          tgtValue = value.toString();
+        }
```

Comment:
Probably loop is not required? `tgtObjFromLocalStorage.TGT`

Reply:
@adongare This is done too.